modules/nodes-descheduler-profiles.adoc

```diff
@@ -0,0 +1,43 @@
+// Module included in the following assemblies:
+//
+// * nodes/scheduling/nodes-descheduler.adoc
+
+[id="nodes-descheduler-profiles_{context}"]
+= Descheduler profiles
+
+The following descheduler profiles are available:
+
+`AffinityAndTaints`:: This profile evicts pods in an effort to evenly spread similar pods, or pods of the same topology domain, among nodes.
++
+It enables the following strategies:
++
+* `RemovePodsViolatingInterPodAntiAffinity`: ensures that pods violating inter-pod anti-affinity are removed from nodes.
+* `RemovePodsViolatingNodeAffinity`: ensures that pods violating node affinity are removed from nodes.
+* `RemovePodsViolatingNodeTaints`: ensures that pods violating `NoSchedule` taints on nodes are removed.
++
+Pods with a node affinity type of `requiredDuringSchedulingIgnoredDuringExecution` are removed.
+
+`TopologyAndDuplicates`:: This profile evicts pods that violate basic pod and node affinity and node taints.
++
+It enables the following strategies:
++
+* `RemovePodsViolatingTopologySpreadConstraint`: finds unbalanced topology domains and tries to evict pods from larger ones when `DoNotSchedule` constraints are violated.
+* `RemoveDuplicates`: ensures that there is only one pod associated with a replica set, replication controller, deployment, or job running on same node. If there are more, then those duplicate pods are evicted for better spreading of pods in a cluster.
+
+`LifecycleAndUtilization`:: This profile evicts long-running pods and balances resource usage between nodes.
++
+It enables the following strategies:
++
+* `RemovePodsHavingTooManyRestarts`: ensures that pods that have been restarted too many times are removed from nodes.
++
+Pods that are restarted more than 100 times are removed. This includes restarts for Init Containers.
+
+* `LowNodeUtilization`: finds nodes that are underutilized and evicts pods, if possible, from overutilized nodes in the hope that recreation of evicted pods will be scheduled on these underutilized nodes.
++
+A node is considered underutilized if its CPU usage is less than 20%, its memory usage is less than 20%, and its number of pods is less than 20% of the allocated values.
++
```

Comment:
@bergerhoffer may be we should rephrase it as below, If a node’s usage is below the configured thresholds for all parameters (CPU, memory, and number of pods), then the node is considered to be underutilized.

Reply:
Yikes, good catch! I changed the order at some point and must have mixed up the descriptions. Will fix!

Reply:
@kasturinarra I was thinking that we should specify what those thresholds were, since users might want to know and they're not configurable or visible. What do you think?

Reply:
@bergerhoffer yes, specifying thresholds is fine along with the above line, not sure how to phrase it so just put that alone. What i wanted to convey a node is set to be underutilized, only if it's usage is below the configured thresholds for all parameters .... Similarly for over utilized as well as i remember we specifically phrased these sentences in the last release

Reply:
@kasturinarra I reworded it again to make it clearer about "all" for underutilization and "any" for "overutilization". Let me know if this looks better or if you have any other feedback. Thanks!